### PR TITLE
Support GUID CNI runtime config

### PR DESCRIPTION
**Pull request by adrianchiris**
we should first:

1. Merge infiniband-guid as runtime config in multi-net-spec
2.  Update `k8snetworkplumbingwg/network-attachment-definition-client/pkg/apis/k8s.cni.cncf.io/v1/types.go` with the new infiniband-guid runtime config. [PR 25](https://github.com/k8snetworkplumbingwg/network-attachment-definition-client/pull/25)
3. update PR go.mod to use a newer version of net-attach-def client from 2

### CODE REVIEW

**Comment by adrianchiris** [comment]
All Prereq have been met.

This PR should ideally be split into 2:
- first 2 commits as general improvements
- last 2 commits for ib guid support

**Comment by adrianchiris** [comment]
The change was validated and confirmed to be working.

File: README.md [hunk 1]
```diff
@@ -87,6 +87,7 @@ metadata:
   namespace: kube-system
 data:
   DAEMON_SM_PLUGIN: "ufm" # Name of the subnet manager plugin
+  DAEMON_SM_PLUGIN_PATH: "/plugins" # Path to SM plugins folder
```
Comment (moshe010) [inline README.md:90]: where do you use it? I still see envDefault:"/plugins"

File: pkg/daemon/daemon.go [hunk 2]
```diff
@@ -246,7 +248,9 @@ func (d *daemon) AddPeriodicUpdate() {
 					d.guidPodNetworkMap[allocatedGUID] = podNetworkID
 				}
 
-				if err = utils.SetPodNetworkGUID(network, allocatedGUID); err != nil {
+				// TODO(adrianc): should we just set GUID both in CNI ARGS and runtime config ?
```
Comment (moshe010) [inline pkg/daemon/daemon.go:251]: why it should be both?

File: pkg/daemon/daemon.go [hunk 3]
```diff
@@ -263,11 +267,12 @@ func (d *daemon) AddPeriodicUpdate() {
 				pod.Annotations[v1.NetworkAttachmentAnnot] = string(netAnnotations)
 			}
 
-			// used GUID as net.HardwareAddress to use it in sm plugin which receive n[]et.HardwareAddress as parameter
+			// Create a GUID list as net.HardwareAddress to be used it in sm plugin
```
Comment (moshe010) [inline pkg/daemon/daemon.go:270]: I think you can remove the "it"

File: pkg/daemon/daemon.go [hunk 4]
```diff
@@ -263,11 +267,12 @@ func (d *daemon) AddPeriodicUpdate() {
...
 			guidList = append(guidList, guidAddr.HardWareAddress())
 			passedPods = append(passedPods, pod)
 		}
 
+		// Get configured PKEY for network and Add the relevand POD GUIDs as members of the PKey via Subnet Manager
```
Comment (moshe010) [inline pkg/daemon/daemon.go:275]: s/Add/add
s/relevand/relevant

File: README.md (hunk 1 above)
Comment (adrianchiris) [reply README.md:90]: This provides means to override the default plugins path.
its shown explicitly in the README and points to the plugins location inside the container image.
as seen [here](https://github.com/Mellanox/ib-kubernetes/blob/c82060e00b9b2e405ba1fc4a6886c59a5789372d/Dockerfile#L15).

This makes life easier when doing development, where i can just point this to my plugin build dir location.
(when not building image or deploying through k8s but invoking binary directly instead)

File: pkg/daemon/daemon.go (hunk 3 above)
Comment (adrianchiris) [reply pkg/daemon/daemon.go:270]: will remove sure

File: README.md (hunk 1 above)
Comment (moshe010) [reply README.md:90]: so why not move "/plugins" to constant as you used it in envDefault as well?

File: pkg/daemon/daemon.go (hunk 2 above)
Comment (adrianchiris) [reply pkg/daemon/daemon.go:251]: i was thinking at the time of a cluster with different versions of ib-sriov cni where some might support infinibandGUID runtime config and some might not, however we should not hide configuration issues here so ill remove this TODO.

File: README.md (hunk 1 above)
Comment (adrianchiris) [reply README.md:90]: This is a README , what do you mean by constant ?
in the code its listed once.

**Comment by adrianchiris** [comment]
I will also open an issue to deprecate the GUID attribute in ib-sriov-cni so we use only one form

File: pkg/daemon/daemon.go (hunk 4 above)
Comment (adrianchiris) [reply pkg/daemon/daemon.go]: Done

File: README.md (hunk 1 above)
Comment (moshe010) [reply README.md:90]: I am talking about pkg/config/config.go  line 19

Comment (adrianchiris) [reply README.md:90]: correction to my above response, i dont think its possible to use variables as input to envDefault
as its a string annotation that `caarlos0/env/v6` use

https://github.com/caarlos0/env#supported-types-and-defaults

Comment (moshe010) [reply README.md:90]: you are right

**Pull request merged**